backend/handlers/institution_parent_request_collection_handler.py

```diff
@@ -54,12 +55,23 @@ def post(self, user, institution_key):
             EntityException
         )
 
+        child_key = data.get('institution_key')
+        child_key = ndb.Key(urlsafe=child_key)
+        requested_inst_key = data.get('institution_requested_key')
+        requested_inst_key = ndb.Key(urlsafe=requested_inst_key)
+
+        Utils._assert(
+            Institution.has_connection_between(requested_inst_key, child_key),
```

Comment:
here you pass parent and child, but the order in the function is the opposite
`def has_connection_between(possible_child_key, possible_parent_key):`

Reply:
As this is the handler responsible for request a parent, it will create a hierarchical loop if the requested institution ( (requested_inst_key)) is a child of the institution that created the request (child_key). Besides that, the method has_connection_between makes a recursion ascending the hierarchy. That's why it is necessary to call this method with the requested_inst_key as the possible_child_key and the child_key as the possible_parent_key.